hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3644,6 +3645,10 @@ func (r *HostedClusterReconciler) validateConfigAndClusterCapabilities(ctx conte
 		errs = append(errs, err)
 	}
 
+	if disconnected, err := isDisconnected(hc); err != nil && disconnected {
```

Comment:
I don't see why we would want to impose this behaviour limiting consumer choices.
I think for now we should just document clearly the implications of deploying the catalog management/guest side in disconnected environments so customers can choose and we develop usage feedback.
If we want we can let the cli have some defaulting opinions for it without restriciting the API.

Reply:
> I don't see why we would want to impose this behaviour limiting consumer choices

(IMHO), it's ugly to have some pods falling on disconnected envs. If we don't support the "mode transition" between `Guest` and `Management` we need at least to keep HCP clean of failing pods.

>  Also I think having our backend to override user intent...

Agree with that but we are setting this by default in the API, so we don't know when it will come from the user or defaulted by the API (with the current implementation). We could implement as you mention and arg in the CLI giving more flexibility to the user but even with that, but the root issue will be there.

We can discuss it on a grooming or team meeting, but IMHO (again) we need to keep the HCP clean of failing pods.